
<!DOCTYPE html>
<html lang='en'>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    <title>Tree - rpms/nmap - src.fedoraproject.org</title>
        <link rel="shortcut icon" type="image/vnd.microsoft.icon"
        href="/theme/static/favicon.ico?version=5.13.3"/>
    <link href="/theme/static/fedora-bootstrap-1.3.0/fedora-bootstrap.min.css?version=5.13.3"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/fonts/fonts.css?version=5.13.3"
        rel="stylesheet" type="text/css" />
    <link href="/theme/static/fonts/hack_fonts/css/hack-extended.min.css?version=5.13.3"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/theme.css?version=5.13.3"
        type="text/css" rel="stylesheet" />

    <link type="text/css" rel="stylesheet" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" href="/static/vendor/font-awesome/font-awesome.css?version=5.13.3"/>
    <link type="text/css" rel="stylesheet" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" href="/static/pagure.css?version=5.13.3"/>
<link nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" rel="stylesheet" href="/static/vendor/highlight.js/styles/github.css?version=5.13.3"/>
<link nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" rel="stylesheet" href="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.css?version=5.13.3"/>
<style nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N">
  .hljs {
    background: #fff;
  }
</style>
  </head>
  <body id="home">
    
    <!-- start masthead -->
    <nav class="navbar navbar-light masthead p-0 navbar-expand">
      <div class="container">
        <a href="/" class="navbar-brand">
        <img height="40" src="/theme/static/pagure-logo.png?version=5.13.3"
             alt="pagure Logo" id="pagureLogo"/>
        </a>
        <ul class="navbar-nav ml-auto">

          

          <li class="nav-item">
            <a class="btn btn-primary" href="/login/?next=https://src.fedoraproject.org/rpms/nmap/blob/f35/f/nmap-7.93-openssl-1.1-build.patch">Log In</a>
          </li>
        </ul>
      </div>
    </nav>
    <!-- close masthead-->

    <div class="bodycontent">


<div class="bg-light border border-bottom pt-3">
  <div class="container">
    <div class="row mb-3">
      <div class="col-6">
        <div class="row">
        <div class="col-auto pr-0">
            <h3>
<i class="fa fa-archive text-muted"></i></h3>
        </div>
        <div class="col-auto pl-2">
            <h3 class="mb-0">
<a href="/projects/rpms/%2A">rpms</a>&nbsp;/&nbsp;<a href="/rpms/nmap"><strong>nmap</strong></a>
            </h3>
        </div>
        </div>
      </div>
      <div class="col-6 text-right">
        <div class="btn-group">
        <div class="btn-group">
        <a href="#"
            class="btn btn-sm dropdown-toggle btn-outline-primary"
            data-toggle="dropdown" id="watch-button">
          <i class="fa fa-clone fa-fw"></i>
          <span>Clone</span>
        </a>
        <div class="dropdown-menu dropdown-menu-right">
          <div class="m-3" id="source-dropdown">
            <div>
              <h5><strong>Source Code</strong></h5>

              <div class="form-group">
                <div class="input-group input-group-sm">
                  <div class="input-group-prepend"><span class="input-group-text">GIT</span></div>
                  <input class="form-control bg-white select-on-focus" type="text" value="https://src.fedoraproject.org/rpms/nmap.git" readonly>
                </div>
              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  </div>
</div>

<ul class="nav nav-tabs nav-small border-bottom-0">
  <li class="nav-item mr-2 text-dark">
    <a class="nav-link active" href="/rpms/nmap">
        <i class="fa fa-code fa-fw text-muted"></i>
        <span class="d-none d-md-inline">Source</span>
    </a>
  </li>

    <li class="nav-item mr-2 text-dark">
        <a class="nav-link" href="https://bugzilla.redhat.com/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&classification=Fedora&product=Fedora&product=Fedora EPEL&component=nmap">
          <i class="fa fa-fw text-muted fa-exclamation-circle"></i>
          <span class="d-none d-md-inline">Issues&nbsp;</span>
          <span class="fa fa-external-link"></span>
        </a>
    </li>

    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/nmap/pull-requests">
          <i class="fa fa-fw text-muted fa-arrow-circle-down"></i>
          <span class="d-none d-md-inline">Pull Requests&nbsp;</span>
          <span class="badge badge-secondary py-0 d-none d-md-inline">
            2
          </span>
      </a>
    </li>


    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/nmap/stats">
          <i class="fa fa-line-chart fa-fw text-muted"></i>
          <span class="d-none d-md-inline">Stats</span>
      </a>
    </li>


</ul>
  </div>
</div>

<div class="container pt-5 repo-body-container">
  <div class="row">
    <div class="col-2">
<nav class="nav nav-tabs nav-sidetabs flex-column">
  <a class=
      "nav-link nowrap
"
      href="/rpms/nmap">
      <i class="fa fa-home text-muted fa-fw"></i>&nbsp;<span class="d-none d-md-inline">Overview</span>
  </a>
  <a class=
    "nav-link nowrap
 active"
    href="/rpms/nmap/tree/f35">
    <i class="fa fa-file-code-o text-muted fa-fw"></i>&nbsp;Files
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/nmap/commits/f35">
    <i class="fa fa-list-alt text-muted fa-fw" data-glyph="spreadsheet"></i>&nbsp;Commits
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/nmap/branches?branchname=f35">
    <i class="fa fa-random text-muted fa-fw"></i>&nbsp;Branches
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/nmap/forks">
    <i class="fa fa-code-fork text-muted fa-fw"></i>&nbsp;Forks
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/nmap/releases">
    <i class="fa fa-tags text-muted fa-fw"></i>&nbsp;Releases
  </a>

  <div class="col-xs-2 line-height-1"></div>
  <h6>Monitoring status:</h6>
  <div class="btn-group">
    <button title="Monitoring status" class="btn btn-sm btn-outline-primary disabled"
        id="monitoring-button">
      <i id="monitoring-icon" class="fa fa-fw fa-eye"></i>
      <span id="monitoring-label" class="fa fa-circle-o-notch fa-spin fa-1x fa-fw"></span>
    </button>
  </div>

  <div class="col-xs-2 line-height-1"></div>
  <div id="orphan-section" class="pt-3">
      <div class="col-xs-2 line-height-1"></div>

  </div>

  <div class="pt-3">
    <div class="col-xs-2 line-height-1">
    <h6>Bugzilla Assignee:</h6>
      <dl>
        <dt>Fedora: </dt>
        <dd id="fedora_assignee_txt">
        mosvald
        </dd>
        <dt>EPEL: </dt>
        <dd id="epel_assignee_txt">
        mosvald
        </dd>
      </dl>
    </div>
  </div>

  <div class="modal fade" id="modal_assignee" tabindex="-1"
          role="dialog" aria-labelledby="Bugzilla assignee" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Bugzilla Assignee</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
        <form id="change_assignees">
          <div class="modal-body">
            <label for="fedora_assignee">Fedora</label>
            <input title="Default assignee for Fedora in bugzilla - Empty input resets to default"
              class="form-control" name="fedora_assignee" id="fedora_assignee" value="mosvald"/>
            <label for="epel_assignee">EPEL</label>
            <input title="Default assignee for EPEL in bugzilla (if applicable) - Empty input resets to default"
              class="form-control" name="epel_assignee" id="epel_assignee" value="mosvald" />
            <p class="pt-2">
              These two fields allow to specify a different default assignee for ticket opened against
              this package in bugzilla. Note: The EPEL field is
              always displayed for packages in the 'rpms' namespace regardless of whether it
              is used in bugzilla or not.            </p>
          </div>
          <div class="modal-footer">
            <button class="btn btn-secondary" type="button" title="Update bugzilla overrides" id="reset_assignees">
               Reset to defaults
            </button>
            <button class="btn btn-primary" type="submit" title="Update bugzilla overrides" id="update_assignees">
               Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>

  <div class="modal fade" id="modal_orphan" tabindex="-1"
          role="dialog" aria-labelledby="Orphan this package" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Orphan package</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
          <div class="modal-body">
            <div class="form-group">
              <label for="reason_menu">Reason</label>
              <select id="reason_menu" class="form-control">

                <option id="lack_of_time_option_button">Lack of time</option>

                <option id="do_not_use_it_option_button">Do not use it anymore</option>

                <option id="unmaintained_option_button">Unmaintained upstream</option>

                <option id="fails_to_build_option_button">Fails to build from source</option>

                <option id="not_fixed_option_button">Important bug not fixed</option>

                <option id="other_option_button">Other</option>
              </select>
            </div>
          <label for="orphan_reason_info">Additional info</label>
          <input title="Additional info for orphaning reason"
                  class="form-control" name="orphan_reason_info" id="orphan_reason_info" value=""/>
          </div>
          <div class="modal-footer">
            <button class="btn btn-primary" type="submit" title="Orphan this package" id="orphan_button">
               Update
            </button>
          </div>
      </div>
    </div>
  </div>

  <script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N">
    window.addEventListener('load', function() {
      set_up_monitoring = function(status){
        var _label = "Disabled"
        if (status === "monitoring") {
          _label = "Monitoring";
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-with-scratch") {
          _label = "Scratch builds"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else {
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye-slash")
        }

        $("#monitoring-label").text(_label);
        $("#monitoring-label").removeClass("fa fa-circle-o-notch fa-spin fa-1x fa-fw");
      }

      $.ajax({
        url: "/_dg/anitya/rpms/nmap",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          console.log(res);
          set_up_monitoring(res.monitoring)
        }
      });


      $("#reset_assignees").on('click', function(){
        $('#fedora_assignee').val('');
        $('#epel_assignee').val('');
        $("#change_assignees").submit();
        return false;
      });

      $("#change_assignees").on('submit',  function(){
        $('html').css('cursor', 'progress');
        $('#reset_assignees').attr('disabled', true);
        $('#update_assignees').attr('disabled', true);
        $('#update_assignees').text('Updating...');
        $.ajax({
          url: "/_dg/bzoverrides/rpms/nmap",
          type: 'POST',
          dataType: 'json',
          data: {
             'epel_assignee': $('#epel_assignee').val(),
             'fedora_assignee': $('#fedora_assignee').val()
          },
          success: function(res) {
            $("#fedora_assignee_txt").text(res.fedora_assignee);
            $("#epel_assignee_txt").text(res.epel_assignee);
            $('#modal_assignee').modal('hide');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            $('html').css('cursor', 'default');
            console.log("Successfully changed the bugzilla assignees");
            return false;
          },
          error: function(res) {
            var msg = '';
            if(res.responseJSON.errors){
              msg = ': ' + res.responseJSON.errors.join(', ');
            }
            alert("Unable to update the bugzilla assignee(s)" + msg);
            $('html').css('cursor', 'default');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            return false;
          }
        })
        return false;
      });


      $.ajax({
        url: "/_dg/actived/rpms/nmap",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          var _btn = $("#take_orphan_button");
          if (!res.active){
            _btn.off("click");
            _btn.click(function(){
                window.open(
                    "https://pagure.io/releng/new_issue?title="
                    + "Unretire rpms/nmap"
                    + "&amp;template=package_unretiremet");
            });
            _btn.prop( "title", "Package retired - Open a releng ticket to adopt it" );
            _btn.html("Retired");
          }
          _btn.removeClass('disabled');
        }
      });

      $("#orphan_button").click(function(){
        $("#orphan_button").attr("disabled", true);
        $.ajax({
          url: "/_dg/orphan/rpms/nmap",
          type: 'POST',
          dataType: 'json',
          data: {
            'orphan_reason': $('#reason_menu').val(),
            'orphan_reason_info': $('#orphan_reason_info').val()
          },
          success: function(res) {
            $("#point_of_contact_div").html("Package is currently unmaintained");
            $("#orphan_button").attr("disabled", false);
            $('#modal_orphan').modal('hide');
            $('#orphan-section').html('');
          },
          error: function(res) {
            if (res.responseJSON.errors) {
              alert('Unable to orphan the package: ' + res.responseJSON.errors);
            } else {
              alert('Unable to orphan the package: ' + res.responseJSON.error);
            }
            $("#orphan_button").attr("disabled", false);
          }
        });
      });
    });
  </script>

</nav>    </div>
    <div class="col-10">
  <div class="row mb-1">
    <div class="col-sm-6">
    <h3>
      Files
    </h3>
    </div>

    <div class="col-sm-6">
      <div class="float-right">
          <div class="btn-group">
            <a href="#" class="btn btn-outline-light border-secondary text-dark btn-sm dropdown-toggle"
                    data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    <span class="fa fa-random fa-fw"></span> Branch: <span class="font-weight-bold">f35</span>
          </a>
            <div class="dropdown-menu dropdown-menu-right">
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f10">
                      <span class="fa fa-random fa-fw"></span> <span class="">f10</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f11">
                      <span class="fa fa-random fa-fw"></span> <span class="">f11</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f12">
                      <span class="fa fa-random fa-fw"></span> <span class="">f12</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f13">
                      <span class="fa fa-random fa-fw"></span> <span class="">f13</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f14">
                      <span class="fa fa-random fa-fw"></span> <span class="">f14</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f15">
                      <span class="fa fa-random fa-fw"></span> <span class="">f15</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f16">
                      <span class="fa fa-random fa-fw"></span> <span class="">f16</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f17">
                      <span class="fa fa-random fa-fw"></span> <span class="">f17</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f18">
                      <span class="fa fa-random fa-fw"></span> <span class="">f18</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f19">
                      <span class="fa fa-random fa-fw"></span> <span class="">f19</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f20">
                      <span class="fa fa-random fa-fw"></span> <span class="">f20</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f21">
                      <span class="fa fa-random fa-fw"></span> <span class="">f21</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f22">
                      <span class="fa fa-random fa-fw"></span> <span class="">f22</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f23">
                      <span class="fa fa-random fa-fw"></span> <span class="">f23</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f24">
                      <span class="fa fa-random fa-fw"></span> <span class="">f24</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f25">
                      <span class="fa fa-random fa-fw"></span> <span class="">f25</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f26">
                      <span class="fa fa-random fa-fw"></span> <span class="">f26</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f27">
                      <span class="fa fa-random fa-fw"></span> <span class="">f27</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f28">
                      <span class="fa fa-random fa-fw"></span> <span class="">f28</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f29">
                      <span class="fa fa-random fa-fw"></span> <span class="">f29</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f30">
                      <span class="fa fa-random fa-fw"></span> <span class="">f30</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f31">
                      <span class="fa fa-random fa-fw"></span> <span class="">f31</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f32">
                      <span class="fa fa-random fa-fw"></span> <span class="">f32</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f33">
                      <span class="fa fa-random fa-fw"></span> <span class="">f33</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f34">
                      <span class="fa fa-random fa-fw"></span> <span class="">f34</span>
                  </a>
                  <a class="dropdown-item pl-1 active" href="/rpms/nmap/tree/f35">
                      <span class="fa fa-random fa-fw"></span> <span class="font-weight-bold">f35</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f36">
                      <span class="fa fa-random fa-fw"></span> <span class="">f36</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f37">
                      <span class="fa fa-random fa-fw"></span> <span class="">f37</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f38">
                      <span class="fa fa-random fa-fw"></span> <span class="">f38</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f7">
                      <span class="fa fa-random fa-fw"></span> <span class="">f7</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f8">
                      <span class="fa fa-random fa-fw"></span> <span class="">f8</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/f9">
                      <span class="fa fa-random fa-fw"></span> <span class="">f9</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/main">
                      <span class="fa fa-random fa-fw"></span> <span class="">main</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/nmap/tree/rawhide">
                      <span class="fa fa-random fa-fw"></span> <span class="">rawhide</span>
                  </a>
            </div>
          </div>
    </div>
  </div>
  </div>
    <div class="card mb-3">
      <div class="card-header">
        <ol class="breadcrumb p-0 bg-transparent mb-0">
          <li class="breadcrumb-item">
            <a href="/rpms/nmap/tree/f35">
              <span class="fa fa-random">
              </span>&nbsp; f35
            </a>
          </li>
          <li class="active breadcrumb-item">
            <span class="fa fa-file" data-glyph="">
            </span>&nbsp; nmap-7.93-openssl-1.1-build.patch
          </li>
        </ol>
      </div>

  <div class="card-body p-0">
            <div class="bg-light border text-right pr-2">
                <form class="btn btn-sm" method="POST" name="fork_project"
                    action="/fork_edit/rpms/nmap/edit/f35/f/nmap-7.93-openssl-1.1-build.patch">
                    <button class="btn btn-sm btn-secondary fork_project_btn">
                            Fork and Edit
                    </button>
                    
                </form>

                <a class="btn btn-secondary btn-sm" href="/rpms/nmap/blob/f35/f/nmap-7.93-openssl-1.1-build.patch" title="View as blob">Blob</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/nmap/blame/nmap-7.93-openssl-1.1-build.patch?identifier=f35" title="View git blame">Blame</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/nmap/history/nmap-7.93-openssl-1.1-build.patch?identifier=f35" title="View git log for this file">History</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/nmap/raw/f35/f/nmap-7.93-openssl-1.1-build.patch" title="View as raw">Raw</a>
            </div>

        <pre class="syntaxhighlightblock"><code class="lang-diff">From d6bea8dcdee36a3902cece14097993350306f1b6 Mon Sep 17 00:00:00 2001
From: dmiller &lt;dmiller@e0a8ed71-7df4-0310-8962-fdc924857419&gt;
Date: Tue, 6 Sep 2022 22:39:34 +0000
Subject: [PATCH] Build based on OpenSSL version, not API level. Fixes #2516

---
 ncat/http_digest.c        |  2 +-
 ncat/ncat_connect.c       |  4 ++--
 ncat/ncat_ssl.c           |  6 +++---
 ncat/ncat_ssl.h           | 12 ------------
 ncat/test/test-wildcard.c |  4 ++--
 nse_openssl.cc            | 28 +++++++---------------------
 nse_ssl_cert.cc           | 24 ++++++------------------
 nsock/src/nsock_ssl.c     |  4 ++--
 nsock/src/nsock_ssl.h     | 15 +--------------
 9 files changed, 24 insertions(+), 75 deletions(-)

diff --git a/ncat/http_digest.c b/ncat/http_digest.c
index b5f80a920..e6ff99175 100644
--- a/ncat/http_digest.c
+++ b/ncat/http_digest.c
@@ -133,7 +133,7 @@ int http_digest_init_secret(void)
     return 0;
 }
 
-#if OPENSSL_API_LEVEL &lt; 10100
+#if OPENSSL_VERSION_NUMBER &lt; 0x10100000L
 #define EVP_MD_CTX_new EVP_MD_CTX_create
 #define EVP_MD_CTX_free EVP_MD_CTX_destroy
 #endif
diff --git a/ncat/ncat_connect.c b/ncat/ncat_connect.c
index 0e4b50761..3dd3291fc 100644
--- a/ncat/ncat_connect.c
+++ b/ncat/ncat_connect.c
@@ -82,8 +82,8 @@
 #include &lt;openssl/err.h&gt;
 
 /* Deprecated in OpenSSL 3.0 */
-#if OPENSSL_API_LEVEL &gt;= 30000
-#define SSL_get_peer_certificate SSL_get1_peer_certificate
+#if OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
+# define SSL_get_peer_certificate SSL_get1_peer_certificate
 #endif
 #endif
 
diff --git a/ncat/ncat_ssl.c b/ncat/ncat_ssl.c
index 9226b4811..3818bfecc 100644
--- a/ncat/ncat_ssl.c
+++ b/ncat/ncat_ssl.c
@@ -80,7 +80,7 @@
 #define FUNC_ASN1_STRING_data ASN1_STRING_data
 #endif
 
-#if OPENSSL_API_LEVEL &gt;= 30000
+#if OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
 #include &lt;openssl/provider.h&gt;
 /* Deprecated in OpenSSL 3.0 */
 #define SSL_get_peer_certificate SSL_get1_peer_certificate
@@ -117,7 +117,7 @@ SSL_CTX *setup_ssl_listen(void)
     OpenSSL_add_all_algorithms();
     ERR_load_crypto_strings();
     SSL_load_error_strings();
-#elif OPENSSL_API_LEVEL &gt;= 30000
+#elif OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
   if (NULL == OSSL_PROVIDER_load(NULL, &#34;legacy&#34;))
   {
     loguser(&#34;OpenSSL legacy provider failed to load.\n&#34;);
@@ -477,7 +477,7 @@ static int ssl_gen_cert(X509 **cert, EVP_PKEY **key)
     const char *commonName = &#34;localhost&#34;;
     char dNSName[128];
     int rc;
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
     int ret = 0;
     RSA *rsa = NULL;
     BIGNUM *bne = NULL;
diff --git a/ncat/ncat_ssl.h b/ncat/ncat_ssl.h
index fca0b1771..458736e27 100644
--- a/ncat/ncat_ssl.h
+++ b/ncat/ncat_ssl.h
@@ -67,18 +67,6 @@
 #include &lt;openssl/ssl.h&gt;
 #include &lt;openssl/err.h&gt;
 
-/* OPENSSL_API_LEVEL per OpenSSL 3.0: decimal MMmmpp */
-#ifndef OPENSSL_API_LEVEL
-# if OPENSSL_API_COMPAT &lt; 0x900000L
-#  define OPENSSL_API_LEVEL (OPENSSL_API_COMPAT)
-# else
-#  define OPENSSL_API_LEVEL \
-     (((OPENSSL_API_COMPAT &gt;&gt; 28) &amp; 0xF) * 10000  \
-      + ((OPENSSL_API_COMPAT &gt;&gt; 20) &amp; 0xFF) * 100 \
-      + ((OPENSSL_API_COMPAT &gt;&gt; 12) &amp; 0xFF))
-# endif
-#endif
-
 #define NCAT_CA_CERTS_FILE &#34;ca-bundle.crt&#34;
 
 enum {
diff --git a/ncat/test/test-wildcard.c b/ncat/test/test-wildcard.c
index 428ece71c..fe55e1997 100644
--- a/ncat/test/test-wildcard.c
+++ b/ncat/test/test-wildcard.c
@@ -20,7 +20,7 @@ are rejected. The SSL transactions happen over OpenSSL BIO pairs.
 
 #include &#34;ncat_core.h&#34;
 #include &#34;ncat_ssl.h&#34;
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
 #include &lt;openssl/bn.h&gt;
 #endif
 
@@ -294,7 +294,7 @@ stack_err:
 static int gen_cert(X509 **cert, EVP_PKEY **key,
     const struct lstr commonNames[], const struct lstr dNSNames[])
 {
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
     int rc, ret=0;
     RSA *rsa = NULL;
     BIGNUM *bne = NULL;
diff --git a/nse_openssl.cc b/nse_openssl.cc
index 3ee5d73d3..0f5b450e0 100644
--- a/nse_openssl.cc
+++ b/nse_openssl.cc
@@ -20,6 +20,9 @@
 #define FUNC_EVP_CIPHER_CTX_init EVP_CIPHER_CTX_reset
 #define FUNC_EVP_CIPHER_CTX_cleanup EVP_CIPHER_CTX_reset
 #define PASS_EVP_CTX(ctx) (ctx)
+#if OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
+# include &lt;openssl/provider.h&gt;
+#endif
 #else
 #define FUNC_EVP_MD_CTX_init EVP_MD_CTX_init
 #define FUNC_EVP_MD_CTX_cleanup EVP_MD_CTX_cleanup
@@ -37,23 +40,6 @@ extern NmapOps o;
 
 #include &#34;nse_openssl.h&#34;
 
-/* OPENSSL_API_LEVEL per OpenSSL 3.0: decimal MMmmpp */
-#ifndef OPENSSL_API_LEVEL
-# if OPENSSL_API_COMPAT &lt; 0x900000L
-#  define OPENSSL_API_LEVEL (OPENSSL_API_COMPAT)
-# else
-#  define OPENSSL_API_LEVEL \
-     (((OPENSSL_API_COMPAT &gt;&gt; 28) &amp; 0xF) * 10000  \
-      + ((OPENSSL_API_COMPAT &gt;&gt; 20) &amp; 0xFF) * 100 \
-      + ((OPENSSL_API_COMPAT &gt;&gt; 12) &amp; 0xFF))
-# endif
-#endif
-
-
-#if OPENSSL_API_LEVEL &gt;= 30000
-#include &lt;openssl/provider.h&gt;
-#endif
-
 #define NSE_SSL_LUA_ERR(_L) \
     luaL_error(_L, &#34;OpenSSL error: %s&#34;, ERR_error_string(ERR_get_error(), NULL))
 
@@ -184,7 +170,7 @@ static int l_bignum_is_prime( lua_State *L ) /** bignum_is_prime( BIGNUM p ) */
   bignum_data_t * p = (bignum_data_t *) luaL_checkudata( L, 1, &#34;BIGNUM&#34; );
   BN_CTX * ctx = BN_CTX_new();
   int is_prime =
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
     BN_is_prime_ex( p-&gt;bn, BN_prime_checks, ctx, NULL );
 #else
     BN_check_prime( p-&gt;bn, ctx, NULL );
@@ -199,7 +185,7 @@ static int l_bignum_is_safe_prime( lua_State *L ) /** bignum_is_safe_prime( BIGN
   bignum_data_t * p = (bignum_data_t *) luaL_checkudata( L, 1, &#34;BIGNUM&#34; );
   BN_CTX * ctx = BN_CTX_new();
   int is_prime =
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
     BN_is_prime_ex( p-&gt;bn, BN_prime_checks, ctx, NULL );
 #else
     BN_check_prime( p-&gt;bn, ctx, NULL );
@@ -210,7 +196,7 @@ static int l_bignum_is_safe_prime( lua_State *L ) /** bignum_is_safe_prime( BIGN
     BN_sub_word( n, (BN_ULONG)1 );
     BN_div_word( n, (BN_ULONG)2 );
     is_safe =
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
       BN_is_prime_ex( n, BN_prime_checks, ctx, NULL );
 #else
       BN_check_prime( n, ctx, NULL );
@@ -582,7 +568,7 @@ LUALIB_API int luaopen_openssl(lua_State *L) {
 #if OPENSSL_VERSION_NUMBER &lt; 0x10100000L || defined LIBRESSL_VERSION_NUMBER
   OpenSSL_add_all_algorithms();
   ERR_load_crypto_strings();
-#elif OPENSSL_API_LEVEL &gt;= 30000
+#elif OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
   if (NULL == OSSL_PROVIDER_load(NULL, &#34;legacy&#34;) &amp;&amp; o.debugging &gt; 1)
   {
     // Legacy provider may not be available.
diff --git a/nse_ssl_cert.cc b/nse_ssl_cert.cc
index 5ae623a47..bc35019a1 100644
--- a/nse_ssl_cert.cc
+++ b/nse_ssl_cert.cc
@@ -89,19 +89,7 @@
 #define X509_get0_notAfter X509_get_notAfter
 #endif
 
-/* OPENSSL_API_LEVEL per OpenSSL 3.0: decimal MMmmpp */
-#ifndef OPENSSL_API_LEVEL
-# if OPENSSL_API_COMPAT &lt; 0x900000L
-#  define OPENSSL_API_LEVEL (OPENSSL_API_COMPAT)
-# else
-#  define OPENSSL_API_LEVEL \
-     (((OPENSSL_API_COMPAT &gt;&gt; 28) &amp; 0xF) * 10000  \
-      + ((OPENSSL_API_COMPAT &gt;&gt; 20) &amp; 0xFF) * 100 \
-      + ((OPENSSL_API_COMPAT &gt;&gt; 12) &amp; 0xFF))
-# endif
-#endif
-
-#if OPENSSL_API_LEVEL &gt;= 30000
+#if OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
 #include &lt;openssl/core_names.h&gt;
 /* Deprecated in OpenSSL 3.0 */
 #define SSL_get_peer_certificate SSL_get1_peer_certificate
@@ -459,7 +447,7 @@ static const char *pkey_type_to_string(int type)
 }
 
 int lua_push_ecdhparams(lua_State *L, EVP_PKEY *pubkey) {
-#if OPENSSL_API_LEVEL &gt;= 30000
+#if OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
   char tmp[64] = {0};
   size_t len = 0;
   /* This structure (ecdhparams.curve_params) comes from tls.lua */
@@ -634,7 +622,7 @@ static int parse_ssl_cert(lua_State *L, X509 *cert)
   else
 #endif
   if (pkey_type == EVP_PKEY_RSA) {
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
     RSA *rsa = EVP_PKEY_get1_RSA(pubkey);
     if (rsa) {
 #endif
@@ -643,7 +631,7 @@ static int parse_ssl_cert(lua_State *L, X509 *cert)
       luaL_getmetatable( L, &#34;BIGNUM&#34; );
       lua_setmetatable( L, -2 );
 #if HAVE_OPAQUE_STRUCTS
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
       const BIGNUM *n = NULL, *e = NULL;
       data-&gt;should_free = false;
       RSA_get0_key(rsa, &amp;n, &amp;e, NULL);
@@ -663,7 +651,7 @@ static int parse_ssl_cert(lua_State *L, X509 *cert)
       luaL_getmetatable( L, &#34;BIGNUM&#34; );
       lua_setmetatable( L, -2 );
 #if HAVE_OPAQUE_STRUCTS
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
       data-&gt;should_free = false;
 #else
       data-&gt;should_free = true;
@@ -673,7 +661,7 @@ static int parse_ssl_cert(lua_State *L, X509 *cert)
       data-&gt;bn = rsa-&gt;n;
 #endif
       lua_setfield(L, -2, &#34;modulus&#34;);
-#if OPENSSL_API_LEVEL &lt; 30000
+#if OPENSSL_VERSION_NUMBER &lt; 0x30000000L
       RSA_free(rsa);
     }
 #endif
diff --git a/nsock/src/nsock_ssl.c b/nsock/src/nsock_ssl.c
index 1ef7d521f..23db5513e 100644
--- a/nsock/src/nsock_ssl.c
+++ b/nsock/src/nsock_ssl.c
@@ -64,7 +64,7 @@
 #include &#34;netutils.h&#34;
 
 #if HAVE_OPENSSL
-#if OPENSSL_API_LEVEL &gt;= 30000
+#if OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
 #include &lt;openssl/provider.h&gt;
 #endif
 
@@ -120,7 +120,7 @@ static SSL_CTX *ssl_init_helper(const SSL_METHOD *method) {
     SSL_library_init();
 #else
     OPENSSL_atexit(nsock_ssl_atexit);
-#if OPENSSL_API_LEVEL &gt;= 30000
+#if OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
     if (NULL == OSSL_PROVIDER_load(NULL, &#34;legacy&#34;))
     {
       nsock_log_error(&#34;OpenSSL legacy provider failed to load.\n&#34;);
diff --git a/nsock/src/nsock_ssl.h b/nsock/src/nsock_ssl.h
index bb99b1b5e..1af473d62 100644
--- a/nsock/src/nsock_ssl.h
+++ b/nsock/src/nsock_ssl.h
@@ -69,20 +69,7 @@
 #include &lt;openssl/err.h&gt;
 #include &lt;openssl/rand.h&gt;
 
-/* OPENSSL_API_LEVEL per OpenSSL 3.0: decimal MMmmpp */
-#ifndef OPENSSL_API_LEVEL
-# if OPENSSL_API_COMPAT &lt; 0x900000L
-#  define OPENSSL_API_LEVEL (OPENSSL_API_COMPAT)
-# else
-#  define OPENSSL_API_LEVEL \
-     (((OPENSSL_API_COMPAT &gt;&gt; 28) &amp; 0xF) * 10000  \
-      + ((OPENSSL_API_COMPAT &gt;&gt; 20) &amp; 0xFF) * 100 \
-      + ((OPENSSL_API_COMPAT &gt;&gt; 12) &amp; 0xFF))
-# endif
-#endif
-
-
-#if OPENSSL_API_LEVEL &gt;= 30000
+#if OPENSSL_VERSION_NUMBER &gt;= 0x30000000L
 /* Deprecated in OpenSSL 3.0 */
 #define SSL_get_peer_certificate SSL_get1_peer_certificate
 #endif
-- 
2.37.3

</code></pre>
  </div>
 </div> <!-- end .card-->

</div>
</div>
</div>
    </div>

        <div class="footer pt-4 text-white">
        <div class="container">
            <div class="d-flex align-items-center">
                <div>
                    <div>Powered by <a href="https://pagure.io/pagure" class="notblue">Pagure</a> 5.13.3</div>
                    <div>
                        <a href="https://docs.pagure.org/pagure/usage/index.html" class="notblue">Documentation</a> &bull;
                        <a href="https://pagure.io/pagure/new_issue" class="notblue">File an Issue</a> &bull;
                        <a href="/about">About this Instance</a> &bull;
                        <a href="/ssh_info" class="notblue">SSH Hostkey/Fingerprint</a>
                    </div>
                </div>
                <div class="ml-auto text-right">
                    <div>&copy; Red Hat, Inc. and others.</div>
                </div>
            </div>
        </div>
    </div>


    <script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" src="/static/vendor/jquery/jquery.min.js?version=5.13.3"></script>

    <script src="/static/vendor/bootstrap/bootstrap.bundle.min.js?version=5.13.3"></script>

    <script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N">
      $('[data-toggle="tooltip"]').tooltip({placement : 'bottom'});
      $(".cancel_btn").click(function() {
        history.back();
      });
    </script>

<script type="text/javascript"  nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" src="/static/vendor/lazyload/lazyload.min.js?version=5.13.3"></script>

<script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N">
window.addEventListener("load", function(event) {
    lazyload();
});
</script>

<script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N">
$("#giturl-toggle").on('click', function(event){
  event.stopPropagation();
  $("#giturl-more").toggle();
  $("#giturl-toggle").hide();
})

$(".fork_project_btn").click(function() {
  $('#fork_project').submit();
});

$(".select-on-focus").on("focus", function() {
  $(this).select();
});

</script>

<script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" src="/static/vendor/highlight.js/highlight.pack.js?version=5.13.3"></script>
<script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" src="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.js?version=5.13.3"></script>
<script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N" src="/static/vendor/highlight.js/spec.js?version=5.13.3"></script>

<script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N">
  $(document).ready(function() {
  $('.fork_project_btn').click($("[name=fork_project]").submit);

  $('pre.syntaxhighlightblock code').each(function(i, block) {
    hljs.highlightBlock(block);
    hljs.lineNumbersBlock(block);
  });

  var cls = "highlighted-line";
  var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
  if (! isNaN(lines[0]))
  {
    for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
      $('#_' + i).parent().parent().addClass(cls);
    }
    setTimeout(function(){
      $("#_" + lines[0]).get(0).scrollIntoView({behavior: "instant", block: "start", inline: "nearest"});
    }, 50);
  }
});
</script>

<script type="text/javascript" nonce="Zv1kN6z3ML2tALzXNJrRQ6X5N">

  function updateHighlight() {
    var cls = "highlighted-line";
    $('.' + cls).removeClass(cls)
    if (location.hash !== '') {
      var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
      for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
        $('[data-line-number=' + i + ']').closest('tr').addClass(cls);
      }
      return lines;
    }
    return [];
  }
  $(window).on('hashchange', updateHighlight);
  var selected = [];
  $("[data-line-number]").click(function (ev) {
    var line = $(this).attr('data-line-number');
    if (ev.shiftKey) {
      selected = selected.slice(-1).concat(line);
    } else {
      selected = [line];
    }

    var hash = '_' + selected[0];
    if (selected.length === 2) {
      hash = '_' + Math.min(selected[0], selected[1]) + '-' + Math.max(selected[0], selected[1]);
    }
    window.location.hash = hash;
    return false;
  });

</script>


</body>
</html>